src/main/java/org/openrewrite/java/testing/cucumber/RegexToCucumberExpression.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.cucumber;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+public class RegexToCucumberExpression extends Recipe {
+
+    private static final String IO_CUCUMBER_JAVA_STEP_DEFINITION = "io.cucumber.java.*.*";
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesType<>(IO_CUCUMBER_JAVA_STEP_DEFINITION);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace Cucumber-Java step definition regexes with Cucumber expressions.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Strip regex prefix and suffix from step annotation expressions arguments where possible.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new CucumberStepDefinitionBodyVisitor();
+    }
+
+    static final class CucumberStepDefinitionBodyVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration m, ExecutionContext p) {
+            J.MethodDeclaration methodDeclaration = super.visitMethodDeclaration(m, p);
+            return methodDeclaration.withLeadingAnnotations(ListUtils.map(methodDeclaration.getLeadingAnnotations(),
+                    ann -> replaceRegexWithCucumberExpression(methodDeclaration, ann, p)));
+        }
+
+        private static J.Annotation replaceRegexWithCucumberExpression(
```

Comment:
This will transform any annotation having a literal argument matching the `.filter(s -> s.startsWith("^") || s.endsWith("$") || leadingAndTrailingSlash(s))` filter.   Consider using an AnnotationMatcher or checking the type to prevent invalid transformations.